lib/xapi/problem.rb

```diff
@@ -60,6 +60,16 @@ def test_files
       Tests.new(path: language_dir, files: files).test_files
     end
 
+    def zip_problem_files(file: Tempfile.new([slug, '.zip']),
```

Comment:
Let's call this `zip` rather than `zip_problem_files`. We're already in the problem, and `problem.zip` makes enough sense that the rest seems a bit redundant.